lib/common/common/src/flags.rs

```diff
@@ -25,10 +28,34 @@ pub struct FeatureFlags {
     pub use_mutable_id_tracker_without_rocksdb: bool,
 }
 
+impl FeatureFlags {
+    /// Check if no flag is set at all
+    pub fn is_empty(self) -> bool {
+        let FeatureFlags {
+            all,
+            use_new_shard_key_mapping_format,
+            use_mutable_id_tracker_without_rocksdb,
+        } = self;
+        !all && !use_new_shard_key_mapping_format && !use_mutable_id_tracker_without_rocksdb
```

Comment:
If we go with the "apply `all` on init" approach of this PR, I think we should remove the check for `all` here since it doesn't serve any purpose but makes things more complex and might give the wrong feeling of the `all` attribute.